pkg/config/monitoringspec.go

```diff
@@ -0,0 +1,80 @@
+package config
+
+import (
+	"errors"
+
+	integreatlyv1alpha1 "github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type MonitoringSpec struct {
+	Config ProductConfig
+}
+
+func NewMonitoringSpec(config ProductConfig) *MonitoringSpec {
+	return &MonitoringSpec{Config: config}
+}
+
+func (m *MonitoringSpec) GetWatchableCRDs() []runtime.Object {
```

Comment:
@Rajagopalan-Ranganathan @grdryn did we decide on watching ServiceMonitors (across all rhmi namespaces) for changes & requeue?
If yes, should that go here, is is there a better place to put it.

Reply:
@grdryn @david-martin As per the design, we first get a list of namespaces that have ```"monitoring-key=middleware"``` . Then get the corresponding service monitors in each of those name spaces. So IMO we are watching them for changes.  This looks like more a static approach rather then the dynamic one we have now. Correct me if i am wrong.

Reply:
Discussed and agreed that, we will watch for servicemonitors.